model/group.js

```diff
@@ -422,6 +423,71 @@ module.exports = (function () {
     });
   };
 
+
+   /**
+  * ### addWatchers
+  *
+  * `addWatchers` is an asynchronous function that check if given data, users
+  * or admins logins, are correct and transforms it to expected values : unique
+  * identifiers, before saving it to database.
+  *
+  * It takes :
+  *
+  * - `gid` group unique identifier;
+  * - array of users `loginsOrEmails`;
+  * - `callback` function calling with  *error* if error or *null* and the
+  *   updated group otherwise, plus accepted and refused invitations logins or
+  *   emails.
+  *
+  * It takes care of exclusion of admins and users : admin status is a
+  * escalation of user.
+  *
+  * As login list should be exhaustive, it also takes care or reindexing user
+  * local groups.
+  */
+  group.addWatchers = function (gid, loginsOrEmails, callback) {
+ 
+    if (!ld.isString(gid)) {
+      throw new TypeError('BACKEND.ERROR.TYPE.GID_STR');
+    }
+    if (!ld.isArray(loginsOrEmails)) {
+      throw new TypeError('BACKEND.ERROR.TYPE.LOGINS_ARR');
+    }
+    if (!ld.isFunction(callback)) {
+      throw new TypeError('BACKEND.ERROR.TYPE.CALLBACK_FN');
+    }
+
+    var users = userCache.fn.getIdsFromLoginsOrEmails(loginsOrEmails);
+    group.get(gid, function (err, g) {
+      if (err) { return callback(err); }
+      var removed;
+      removed = ld.difference(g.watchers, users.uids);
+        for(var i=0; i<g.watchers.length; i++){
```

Comment:
Add safety check in case g.watchers is null (caused by previously existing records before we introduced watchers)